model-optimizer/extensions/middle/split_tdnn_memoryoffset.py

```diff
@@ -0,0 +1,46 @@
+"""
+ Copyright (C) 2018-2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from mo.graph.graph import Graph
+from mo.middle.replacement import MiddleReplacementPattern
+from mo.ops.memoryoffset import MemoryOffset
+from mo.ops.result import Result
+
+
+class SplitTdnnMemoryOffset(MiddleReplacementPattern):
+    '''
+    Splits MemoryOffsets in TDNN blocks into 2 parts. These parts then will be converted to ReadValue and Assign.
+    '''
+    enabled = True
+    run_not_recursively = True
+
+    def run_after(self):
+        from extensions.middle.PartialInfer import PartialInfer
+        return [PartialInfer]
+
+    def run_before(self):
+        from extensions.middle.ReplaceMemoryOffsetWithSplice import ReplaceMemoryOffsetWithMemoryNodePattern, ReplaceMemoryOffsetNodePattern
+        return [ReplaceMemoryOffsetNodePattern, ReplaceMemoryOffsetWithMemoryNodePattern]
+
+    def find_and_replace_pattern(self, graph: Graph):
+        for offset_node in graph.get_op_nodes(op='MemoryOffset', splitted=False):
```

Comment:
What if `splitted` is `None`? Should the transformation be applied to ir?

Reply:
`splitted` can not be None. At least is should newer be None. Currently there is no case when it can reach to value `None`. I don't think that we need some additional check for this case.

Reply:
If the attribute is not set then it will be basically `None`. And it is not set then it should be considered as `False`. Am I right?

If so then we need this check. Or, you can add a default value for the `splitted` in the operation constructor. In this case you don't need to add any check here.

Reply:
Agree. Added default value of `splitted = False`.